src/ai/backend/storage/volumes/xfs/__init__.py

```diff
@@ -281,6 +287,39 @@ class XfsVolume(BaseVolume):
 
     project_registry: XfsProjectRegistry
 
+    def __init__(
+        self,
+        local_config: Mapping[str, Any],
+        mount_path: Path,
+        *,
+        etcd: AsyncEtcd,
+        event_dispatcher: EventDispatcher,
+        event_producer: EventProducer,
+        watcher: Optional[WatcherClient] = None,
+        options: Optional[Mapping[str, Any]] = None,
+    ) -> None:
+        super().__init__(
+            local_config,
+            mount_path,
+            etcd=etcd,
+            event_dispatcher=event_dispatcher,
+            event_producer=event_producer,
+            watcher=watcher,
+            options=options,
+        )
+        self._lock_path = Path(self.config.get("lock_file_path", DEFAULT_LOCK_FILE))
+        try:
+            self._lock_path.touch()
+        except OSError as e:
+            log.exception(
+                "Failed to create XFS backend lock file at {}: (Error: {})",
+                self._lock_path,
+                e,
+            )
+            raise InvalidConfigError(
+                f"Cannot create XFS backend lock file at {self._lock_path}"
+            ) from e
+
```

Comment:
It is generally discouraged to do work that might cause an error in the constructor.